mb_modelbase/cache/DictCache.py

```diff
@@ -0,0 +1,59 @@
+from mb_modelbase.cache import BaseCache
+import time
+import pickle
+import threading
+
+
+class DictCache(BaseCache):
+    """
+    Cache with a regular python dictionary as a backend.
+    Stores the content after every save_interval to a file on disk
+    Ensures mutual exclusion with a lock
+
+    Attributes:
+        save_path (str): path to where the cache should save the dictionary
+        save_interval (float): interval in seconds on which to save the content of the dictionary
+
+    Todo:
+        Implement cache as a monitor like in the readers writers problem
+        Maybe lift mutual exclusion to superclass?
+    """
+
+    def __init__(self, save_path='/tmp/modelbaseCache', save_interval=30):
```

Comment:
What if folder /tmp does not exist?

Reply:
Alos, add missing documentation for arguments.

Reply:
Added documentation and a check if the directory exists.